### PR TITLE
NP-283: Dynamic tests

### PR DESCRIPTION
**Pull request by Reece-Carruthers**
Create an eori generator  which can create a sequence of authorised and invalid eoris, or a list of only authorised/invalid eoris, as well as tests to ensure functionallity works as expected

### CODE REVIEW

**Comment by Reece-Carruthers** [comment]
Interesting finding of a 500 being returned when a list of 3000 with massive strings was passed to the API
```scala
    Scenario("400 Bad Eoris") {
      Given("a bearer token")
      And("an invalid payload")

      val eoris   = useGarbageGenerator(
        authorisedEoris.size
      ) // TODO Note: Had to limit the garbage generator as it was causing the test to fail when the strings were too long causing a 500, is this something we need to look into?.
      val request = createRequest(localNow, eoris)

      When("post a authorisations request to uknw-auth-checker-api with bearer token")
      val response = checkerApiService.authorisations(Json.toJson(request), authBearerToken)

      val invalidFormatErrors = eoris.map(eori => Errors.InvalidEoriFormat(eori))

      val expectedResponse = EisErrorsResponse(StatusAndMessage.BadRequest, Some(invalidFormatErrors))

      Then("I am returned a status code 400")
      response hasStatusAndBody expectedResponse
    }
    ```

**Comment by Reece-Carruthers** [comment]
How do we now return a 500

```scala
  Feature("500, INTERNAL_SERVER_ERROR case Scenarios") { // TODO: How do we trigger an internal server error in the stub/EIS?
    Scenario("500, internal server error") {
      Given("Valid bearer token")
      And("a valid payload")
      val eoris   = useEoriGenerator(fetchRandomNumber(2, authorisedEoris.size))
      val request = createRequest(localNow, eoris)

      When("a POST authorisations request to uknw-auth-checker-api with bearer token")
      val response = checkerApiService.authorisations(Json.toJson(request), authBearerToken)

      val expectedResponse = EisErrorsResponse(StatusAndMessage.InternalServerError, None)

      Then("I am returned a status code 500")
      response hasStatusAndBody expectedResponse
    }
  }

**Comment by TheDanielDoyle** [comment]
> Interesting finding of a 500 being returned when a list of 3000 with massive strings was passed to the API
> 
> ```scala
>     Scenario("400 Bad Eoris") {
>       Given("a bearer token")
>       And("an invalid payload")
> 
>       val eoris   = useGarbageGenerator(
>         authorisedEoris.size
>       ) // TODO Note: Had to limit the garbage generator as it was causing the test to fail when the strings were too long causing a 500, is this something we need to look into?.
>       val request = createRequest(localNow, eoris)
> 
>       When("post a authorisations request to uknw-auth-checker-api with bearer token")
>       val response = checkerApiService.authorisations(Json.toJson(request), authBearerToken)
> 
>       val invalidFormatErrors = eoris.map(eori => Errors.InvalidEoriFormat(eori))
> 
>       val expectedResponse = EisErrorsResponse(StatusAndMessage.BadRequest, Some(invalidFormatErrors))
> 
>       Then("I am returned a status code 400")
>       response hasStatusAndBody expectedResponse
>     }
>  
> ```

https://www.playframework.com/documentation/3.0.x/ScalaBodyParsers#Max-content-length

File: project/Dependencies.scala [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
-  val test: Seq[ModuleID] = Seq(
-    "org.commonmark"       % "commonmark"              % "0.22.0"      % Test,
-    "com.typesafe"         % "config"                  % "1.4.3"       % Test,
-    "org.playframework"   %% "play-ahc-ws-standalone"  % "3.0.5"       % Test,
-    "org.playframework"   %% "play-ws-standalone-json" % "3.0.5"       % Test,
-    "com.vladsch.flexmark" % "flexmark-all"            % "0.64.8"      % Test,
-    "org.scalatest"       %% "scalatest"               % "3.3.0-SNAP4" % Test,
-    "org.slf4j"            % "slf4j-simple"            % "2.0.16"      % Test
+  private val bootstrapVersion = "9.3.0"
+  val PekkoVersion             = "1.1.0-M1"
```
Comment (TheDanielDoyle) [inline project/Dependencies.scala:5]: This can be removed.

File: src/test/scala/uk/gov/hmrc/api/models/constants/JsonPaths.scala [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+object JsonPaths {
+  val authType: String       = "authType"
+  val code: String           = "code"
+  val eoris: String          = "eoris"
+  val errorDetail: String    = "errorDetail"
+  val errors: String         = "errors"
+  val message: String        = "message"
+  val path: String           = "path"
+  val processingDate: String = "processingDate"
```
Comment (TheDanielDoyle) [inline src/test/scala/uk/gov/hmrc/api/models/constants/JsonPaths.scala:27]: Is this in use?

Comment (Reece-Carruthers) [reply src/test/scala/uk/gov/hmrc/api/models/constants/JsonPaths.scala:27]: Some of them are, I'll remove the unneeded ones

**Review by TheDanielDoyle — APPROVED** [review]

**Pull request merged**